pkg/plugin/plugin.go

```diff
@@ -118,13 +118,27 @@ func setupVeth(contNetns ns.NetNS, contIfaceName string) (*current.Interface, *c
 			return fmt.Errorf("failed to lookup %q: %v", containerVeth.Name, err)
 		}
 
-		// In case the MAC address is already assigned to another interface, retry
 		var containerMac net.HardwareAddr
-		for i := 1; i <= macSetupRetries; i++ {
-			containerMac = generateRandomMac()
-			err = netlink.LinkSetHardwareAddr(containerLink, containerMac)
-			if err != nil && i == macSetupRetries {
-				return fmt.Errorf("failed to set container iface %q MAC %q: %v", containerVeth.Name, containerMac.String(), err)
+		if requestedMac != "" {
+			containerMac, err = net.ParseMAC(requestedMac)
+			if err != nil {
+				return fmt.Errorf("failed to parse requested MAC  %q: %v", requestedMac, err)
+			}
+			err = assginMacToLink(containerLink, containerMac, containerVeth.Name, true)
+			if err != nil {
+				return err
+			}
+		} else {
+			// In case the MAC address is already assigned to another interface, retry
+			for i := 1; i <= macSetupRetries; i++ {
+				if requestedMac == "" {
+					containerMac = generateRandomMac()
+				}
+
+				err = assginMacToLink(containerLink, containerMac, containerVeth.Name, i == macSetupRetries)
+				if err != nil {
```

Comment:
I think we should check for macSetupRetries here instead of using errorOnFailure - assignMacToLink should be simple function